app/javascript/crayons/Pills/Pill.jsx

```diff
@@ -0,0 +1,92 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { useState } from 'preact/hooks';
+import classNames from 'classnames/bind';
+import { Icon } from '@crayons';
+import XIcon from '@images/x.svg';
+
+export const Pill = ({
+  children,
+  descriptionIcon,
+  actionIcon,
+  destructiveActionIcon,
+  className,
+  tooltip,
+  onKeyUp,
+  noAction,
+  ...otherProps
+}) => {
+  const [suppressTooltip, setSuppressTooltip] = useState(false);
+
+  const handleKeyUp = (event) => {
+    onKeyUp?.(event);
+    if (!tooltip) {
+      return;
+    }
+    setSuppressTooltip(event.key === 'Escape');
+  };
+
+  const classes = classNames('c-pill', {
+    'c-pill--description-icon': descriptionIcon,
+    'c-pill--action-icon': actionIcon || destructiveActionIcon,
+    'c-pill--action-icon--destructive': destructiveActionIcon,
+    'crayons-tooltip__activator': tooltip,
+    'cursor-default': noAction,
+    'cursor-help': tooltip && noAction,
+    [className]: className,
+  });
+
+  return (
+    <button
+      className={classes}
+      {...otherProps}
+      type="button"
+      onKeyUp={handleKeyUp}
+      aria-disabled={noAction}
```

Comment:
I don't mind if it's tackled here or when we start using this component in the app, but worth noting that `aria-disabled` doesn't stop the `onClick` being triggered. So we might want to add a a click handler to this button along the lines of

```jsx
onClick={(e) => {
    if (!noAction) {
      props.onClick();
    }
  }
}
```

Reply:
This will become an issue as soon as we start using the component, so might as well fix it now.

Reply:
fixed!